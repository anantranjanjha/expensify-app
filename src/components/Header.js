import React from "react";
import { NavLink } from 'react-router-dom';
import { connect } from "react-redux";
import { startLogOut } from "../actions/auth";

const Header = ({startLogOut}) => (
    <header>
        <h1>Expensify App</h1>
        <NavLink to="/dashboard" activeClassName="is-active">Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">CreateExpense</NavLink>
        <button onClick={startLogOut}>LogOut</button>
    </header>
);

const mapDispatchToProps = (dispatch)=>({
    startLogOut: ()=> dispatch(startLogOut())
})


export default connect(undefined,mapDispatchToProps)(Header);