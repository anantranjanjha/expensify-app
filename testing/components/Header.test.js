import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../src/components/Header';

test('Testing header componnents',()=>{
    const wrapper=shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
//     const renderer= new ShallowRenderer();
//     renderer.render(<Header />);
//    expect(renderer.getRenderOutput()).toMatchSnapshot();
})