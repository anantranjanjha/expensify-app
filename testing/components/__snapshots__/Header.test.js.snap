// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing header componnents 1`] = `
<div>
  <h1>
    Expensify App
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Create
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/edit"
  >
    Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
    Help
  </NavLink>
</div>
`;
